api/internal/plugins/loader/loader.go

```diff
@@ -95,13 +100,47 @@ func relativePluginPath(id resid.ResId) string {
 		strings.ToLower(id.Kind))
 }
 
-func AbsolutePluginPath(pc *types.PluginConfig, id resid.ResId) string {
-	return filepath.Join(
-		pc.AbsPluginHome, relativePluginPath(id), id.Kind)
+func (l *Loader) AbsolutePluginPath(id resid.ResId) (string, error) {
+	pluginHome, err := l.absPluginHome()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(pluginHome, relativePluginPath(id), id.Kind), nil
 }
 
-func (l *Loader) absolutePluginPath(id resid.ResId) string {
-	return AbsolutePluginPath(l.pc, id)
+// absPluginHome is the home of kustomize Exec and Go plugins.
+// Kustomize plugin configuration files are k8s-style objects
+// containing the fields 'apiVersion' and 'kind', e.g.
+//   apiVersion: apps/v1
+//   kind: Deployment
+// kustomize reads plugin configuration data from a file path
+// specified in the 'generators:' or 'transformers:' field of a
+// kustomization file.  For Exec and Go plugins, kustomize
+// uses this data to both locate the plugin and configure it.
+// Each Exec or Go plugin (its code, its tests, its supporting data
+// files, etc.) must be housed in its own directory at
+//   ${absPluginHome}/${pluginApiVersion}/LOWERCASE(${pluginKind})
+// where
+//   - ${absPluginHome} is an absolute path, defined below.
+//   - ${pluginApiVersion} is taken from the plugin config file.
+//   - ${pluginKind} is taken from the plugin config file.
+func (l *Loader) absPluginHome() (string, error) {
```

Comment:
Note: if/when we implement https://github.com/kubernetes-sigs/kustomize/issues/2684, this would need to change to cache multiple valid homes, and AbsolutePluginPath would need to use `FirstDirThatExistsElseError` to find the path for the specific plugin.

Reply:
Ack.

Also maybe this makes more sense as a method on PluginConfig?

Reply:
I agree with that conceptually, and looking through the code I'd also say that `MakePluginConfig`, `DisabledPluginConfig` and `EnabledPluginConfig` should probably move to `types` too since they're just constructors for `PluginConfig`. However, attempting this is raising some questions for me about packaging and dependencies. I see that in addition to `types`, we also have a `konfig` package that provides types and constants. What's the intended dividing line there? If I move this "plugin home" logic over to `types` as is, `types` would end up importing `konfig` (which imports `types`, so not ok). If I break that by moving everything relevant from `konfig` into `types`, it's quite a lot of stuff, including some utility functions like `FirstDirThatExistsElseError` (which is where the `types` dependency is coming from). Any advice?

Reply:
Stacking packages from least dependent on other kustomize packages you get roughly, 

`api/filesys` - no kustomize deps, safe to depend on.

`api/resid` has no kustomize deps (not counting kyaml as kustomize).  It's a basic package with basic types `Gvk` and `ResId`.  Making it depend on other packages would cause cycles.  Safe to depend on.   It predates kyaml.  We could eliminate `resid` in favor of things like the `ResourceIdentifier` in `./kustomize/kyaml/yaml/types.go`

`api/types` - the kustomization struct and things it contains.  A collection of basic structs with no dependence on kustomize (other than `resid`).  Safe to depend on (unless you're `resid`)

`api/konfig` -  holds configuration constants and machinery to make those constants more useful.  Depends on `api/types`.  It has, for example, the legacy field paths that we'd ultimately want to get from openapi.

Obviously, this is organic. A package called `types` might as well be called `misc` or `kitchensink` - it's a grab of relatively low level things.  `api/resource` and `api/resmap` are higher up.

I'll write this up into an ARCHITECTURE doc.

Let's just merge this PR to fix the problem, and refactor in some dedicated PR.

Reply:
Thanks for the overview; that's very helpful!